### PR TITLE
Add xml comments to IDocumentExecuter

### PR DESCRIPTION
**Pull request by Shane32**
See 'files changed' for suggested change

### CODE REVIEW

**Comment by sungam3r** [comment]
What is the purpose of this PR? I remember that @joemcbride was negative about excessive xml commenting. See #369:

> In many cases, the XML docs just state what is already there and little effort is put into adding the knowledge and context that needed to understand what is actually going on. I can figure out a method signature easily enough.

**Comment by Shane32** [comment]
I am requesting limited XML comments to the public interfaces (and data-only classes where prudent) so that their purpose can be clarified.  This does not clutter any code, per se, and has a benefit that any classes that inherit these methods will inherit their XML comments as well.  It should be extremely useful.  I do not think this goes against anything @joemcbride stated in #369 including his statement (taken out of context):

> I think XML docs could help though won't necessarily solve the problem of knowing which one to use and why, without perhaps adding tons of docs in the XML comments which I think I would be opposed to.

We can asses the changes on a case-by-case basis, as I will post separate PRs for separate files.  Builders perhaps are not such a good candidate, as they may be well documented on the site.  But it will really help people understand the "internal workings" of graphql-dotnet if the interfaces required for its core operation (ISchema, IDocumentExecuter, etc) have xml comments.  For example, what's the difference between a IDocumentExecuter and a IExecutionStrategy?  They both have a single method ExecuteAsync, and both return a ExecutionResult.  There is no documentation for this.  You and I may know what they do, but a few xml comments will sufficiently document this for the project to help others understand.

**Comment by sungam3r** [comment]
> For example, what's the difference between a IDocumentExecuter and a IExecutionStrategy? They both have a single method ExecuteAsync, and both return a ExecutionResult

Ok. I read the comments on the methods of both interfaces, but still did not understand the difference. if you add documentation, it should be meaningful in order to be useful.

**Comment by Shane32** [comment]
Agreed

**Comment by sungam3r** [comment]
Here it would be possible to describe the areas of their responsibility and the relationship between them (`IDocumentExecuter` calls `IExecutionStrategy` as one of steps, etc)

**Comment by Shane32** [comment]
@sungam3r Please provide comments to the suggested changes.  I'd be happy to work with you on suggested text, wording, amount of detail, etc.  I do feel some documentation in this area is valuable to the project.

File: src/GraphQL/Execution/IDocumentExecuter.cs [hunk 1]
```diff
@@ -3,10 +3,26 @@
 
 namespace GraphQL
 {
+    /// <summary>
+    /// Processes an entire GraphQL request, given an input GraphQL request string.  This is intended to
```
Comment (sungam3r) [inline src/GraphQL/Execution/IDocumentExecuter.cs]:
```suggestion
    /// Processes an entire GraphQL request, given an input GraphQL request string. This is intended to
```

File: src/GraphQL/Execution/IDocumentExecuter.cs [hunk 2]
```diff
@@ -3,10 +3,26 @@
...
+    /// be called by user code to process a query.<br/>
+    /// <br/>
+    /// Typical implementation starts metrics if enabled, relies on IDocumentBuilder to parse the query,
+    /// executes document listeners if attached, selects an execution strategy, and executes the query
+    /// via IExecutionStrategy.  Unhandled exceptions are handled as appropriate for the selected options.
```
Comment (sungam3r) [inline src/GraphQL/Execution/IDocumentExecuter.cs]:
```suggestion
    /// via IExecutionStrategy. Unhandled exceptions are handled as appropriate for the selected options.
```

**Comment by sungam3r** [comment]
Now your comment makes much more sense. Now it’s clear not only what’s happening from it, but also that this interface is intended to be called by user code. If you intend to continue to add documentation (this PR or others), then I recommend using `<see cref="XXX"/>` notation instead of just `XXX`.

**Pull request merged**